lib/off_broadway_amqp10/amqp_10_client/client_impl.ex

```diff
@@ -35,8 +46,16 @@ defmodule OffBroadwayAmqp10.Amqp10.Client.Impl do
 
   @impl Client
   def body(raw_msg) do
-    [payload] = :amqp10_msg.body(raw_msg)
-    payload
+    body = :amqp10_msg.body(raw_msg)
+
+    cond do
+      match?([_], body) ->
+        List.first(body)
+
+      Record.is_record(body, @amqp_value_record_tag) ->
+        packed_content = amqp_value(body, :content)
+        :amqp10_client_types.unpack(packed_content)
```

Comment:
What if we raise when there's `[#'v1_0.amqp_sequence'{}]`?
Currently it would go into the first match.

One way of doing that would be to check if the content is a binary.